webapp/lib/view.dart

```diff
@@ -20,6 +20,10 @@ void init() {
   contentView = new ContentView();
   headerElement.append(navView.navViewElement);
   mainElement.append(contentView.contentViewElement);
+  navView.navActions = {
+      controller.NavAction.allProjects: navView.setAllProjectsLinkVisibility,
+      controller.NavAction.dashboard: navView.setDashboardLinkVisibility
+  };
```

Comment:
as per comment above - this should be refactored into a method, something like:

```
void showParent(NavAction action) {
  switch (action) {
    case NavAction.allProjects:
      _allProjectsLink.classes.toggle('nav-links__link--show', show);
      break;
    .....
```

Reply:
Re-implemented